nokogiri/mediaflux.rb

```diff
@@ -0,0 +1,92 @@
+require 'net/http'
+require 'rexml/document'
+require 'nokogiri'
+
+module MediaFlux
+
+  class MFError < StandardError
+    attr_reader :error, :message, :stack
+    def initialize(response_doc)
+      reply = response_doc.elements['/response/reply']
+      @error = reply.elements['error'].text
+      @message = reply.elements['message'].text
+      @stack = reply.elements['stack']&.text
+    end
+  end
+
+  class MFClient
```

Comment:
I would have to say for longer term understanding, this class is much more readable.  That said the example code is less readable.  It is a toss up, if our only two choices are Nokogiri or bake our own.